src/factories/flowRunState.ts

```diff
@@ -0,0 +1,107 @@
+import { ColorSource, Container } from 'pixi.js'
+import { rectangleFactory } from '@/factories/rectangle'
+import { StateEvent } from '@/models/states'
+import { waitForApplication, waitForViewport } from '@/objects'
+import { waitForConfig } from '@/objects/config'
+import { emitter } from '@/objects/events'
+import { waitForScale } from '@/objects/scale'
+import { layout } from '@/objects/settings'
+
+export type FlowRunStateFactory = Awaited<ReturnType<typeof flowRunStateFactory>>
+
+type FlowRunStateFactoryOptions = {
+  end: Date,
+}
+
+type StateRectangleRenderProps = {
+  x: number,
+  width: number,
+  background: ColorSource,
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function flowRunStateFactory(state: StateEvent, options?: FlowRunStateFactoryOptions) {
+  const application = await waitForApplication()
+  const viewport = await waitForViewport()
+  const config = await waitForConfig()
+  let scale = await waitForScale()
+
+  const element = new Container()
+  const bar = await rectangleFactory()
+  const area = await rectangleFactory()
```

Comment:
I love that we have these primitive shape factories that we can just reuse again and again.